src/platform_impl/linux/wayland/pointer.rs

```diff
@@ -204,27 +205,30 @@ pub fn implement_pointer(
                         let axis_discrete_buffer = axis_discrete_buffer.take();
                         if let Some(surface) = mouse_focus.as_ref() {
                             let wid = make_wid(surface);
+                            let scale_factor = surface::get_dpi_factor(&surface) as f64;
                             if let Some((x, y)) = axis_discrete_buffer {
+                                let delta = LogicalPosition::new(x as f64, y as f64)
+                                    .to_physical(scale_factor);
                                 sink.send_window_event(
                                     WindowEvent::MouseWheel {
                                         device_id: crate::event::DeviceId(
                                             crate::platform_impl::DeviceId::Wayland(DeviceId),
                                         ),
-                                        delta: MouseScrollDelta::LineDelta(x as f32, y as f32),
+                                        delta: MouseScrollDelta::LineDelta(delta),
```

Comment:
We shouldn't scale `LineDelta` since it's a discrete value under the hood, which does require it. So add a comment about it and pass it right away.